### PR TITLE
Just toplevel features supporting locating and reading files for embedding in classic

### PR DESCRIPTION
**Pull request by prozacchiwawa**
relates to https://github.com/Chia-Network/clvm_tools_rs/pull/123

### CODE REVIEW

File: src/classic/clvm_tools/stages/stage_2/operators.rs [hunk 1]
```diff
@@ -52,6 +52,36 @@ pub struct CompilerOperatorsInternal {
     opt_memo: RefCell<HashMap<AllocatorRefOrTreeHash, NodePtr>>,
 }
 
+/// Given a list of search paths, find a full path to a file whose partial name
+/// is given.  If the file can't be found in any search path, use the expression
+/// the user gave to cause the file to be searched for in the error result.
```
Comment (aqk) [inline src/classic/clvm_tools/stages/stage_2/operators.rs:57]: Suggestion: Comment should state what happens if multiple paths match

File: src/classic/clvm_tools/stages/stage_2/operators.rs [hunk 2]
```diff
@@ -52,6 +52,36 @@ pub struct CompilerOperatorsInternal {
...
+        let f_path = path_buf.as_path();
+        if f_path.exists() {
+            return f_path
+                .to_str()
+                .map(|x| x.to_owned())
+                .map(Ok)
+                .unwrap_or_else(|| {
+                    Err(EvalErr(
+                        parent_sexp,
+                        "could not compute absolute path".to_string(),
```
Comment (aqk) [inline src/classic/clvm_tools/stages/stage_2/operators.rs]: Suggestion: This error message should include `f_path` in the message

File: src/tests/classic/stage_2.rs [hunk 3]
```diff
@@ -133,3 +136,48 @@ fn test_compile_assert_2() {
         "(a (q \"opt\" (q 2 (\"opt\" (\"com\" (q \"assert\" 1) (q (\"assert\" (a (i 3 (q 4 (q . 26982) (c 2 (c (c (q . \"assert\") 3) (q (x))))) (q . 2)) 1)) (26982 (c (q . 2) (c (c (q . 3) (c 2 (c (c (q . \"function\") (c 5 ())) (c (c (q . \"function\") (c 11 ())) ())))) (q 64)))) (\"function\" (c (q . \"opt\") (c (c (q . \"com\") (c (c (q . 1) 2) (q (29041 (\"unquote\" (\"macros\"))) (29041 (\"unquote\" (\"symbols\")))))) ()))) (\"list\" (a (q 2 (q 2 2 (c 2 (c 3 (q)))) (c (q 2 (i 5 (q 4 (q . 4) (c 9 (c (a 2 (c 2 (c 13 (q)))) (q)))) (q 1)) 1) 1)) 1)) (\"defmacro\" (c (q . \"list\") (c (f 1) (c (c (q . \"mod\") (c (f (r 1)) (c (f (r (r 1))) (q)))) (q)))))) (q))) 1)) 1)".to_string()
     );
 }
+
```
Comment (aqk) [inline src/tests/classic/stage_2.rs:140]: Great tests. Another case that might be helpful is to test the case where an include file exists, but not in the place you expect (e.g. the command line says to look for it in a subdirectory, but it is in our current working directory (or in both places).

This type fo test just brings the behaviour from documentation into the foreground.

File: src/classic/clvm_tools/stages/stage_2/operators.rs (hunk 1 above)
Comment (prozacchiwawa) [reply src/classic/clvm_tools/stages/stage_2/operators.rs:57]: ok

File: src/classic/clvm_tools/stages/stage_2/operators.rs (hunk 2 above)
Comment (prozacchiwawa) [reply src/classic/clvm_tools/stages/stage_2/operators.rs]: makes sense

File: src/tests/classic/stage_2.rs (hunk 3 above)
Comment (prozacchiwawa) [reply src/tests/classic/stage_2.rs:140]: hmm ok

**Review by aqk — APPROVED** [review]

**Pull request merged**